docs/DeveloperGuide.md

```diff
@@ -1,38 +1,460 @@
+   calling `TableFormatter#addRow` for each group to create a table with the summary of each group. A `String` object
+   representing the table is then returned.
+6. The `String` object retrieved is printed out with `TextUI#printlnMessage`.
 
 ## Product scope
 ### Target user profile
-
-{Describe the target user profile}
+* Budget conscious.
+* Prefers to use desktop applications over other types of applications.
+* Are able to type relatively fast.
+* Comfortable using CLI applications.
 
 ### Value proposition
-
-{Describe the value proposition: what problem does it solve?}
+SplitLah will help the user keep track of expenditures made during each group outing that they go on and help the user 
+calculate what transactions that they should make in order to ensure that everyone pays equally for the activities that 
+they engage in during the outings.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...              | I want to ...                        | So that I can ...                                                           |
+|---------|-----------------------|--------------------------------------|-----------------------------------------------------------------------------|
+| v1.0    | New user              | see usage instructions               | refer to them when I forget how to use the application                      |
+| v1.0    | Budget conscious user | create sessions                      | record the transactions of the outing                                       |
+| v1.0    | Budget conscious user | create activities                    | able to track the expenditure for each activity                             |
+| v1.0    | Budget conscious user | view an existing session             | view the full details of a session previously created                       |
+| v1.0    | Budget conscious user | view an existing activity            | view the cost breakdown of who paid and who was involved in the activity    |
+| v1.0    | Budget conscious user | list all existing sessions           | view all sessions previously created                                        |
+| v1.0    | Budget conscious user | list all activities in a session     | view all the activities that happened in the session                        |
+| v1.0    | Budget conscious user | settle all transactions of a session | see a summary of who needs to pay what amount to who for the entire session |
+| v1.0    | User                  | exit the application                 | stop tracking                                                               |
```

Comment:
Do add in v2.0 and can consider having a separate table for each user type or sort them by different types of user